src/Admin.cpp

```diff
@@ -100,23 +143,118 @@ namespace geopm
 
     std::string Admin::whitelist(int cpuid)
     {
-        std::string result;
         if (cpuid == -1) {
-           if (m_msriog == nullptr) {
-               m_msriog = std::make_shared<MSRIOGroup>();
-           }
-           result = m_msriog->msr_whitelist();
+            cpuid = m_cpuid_local;
         }
-        else {
-            result = geopm::MSRIOGroup::msr_whitelist(cpuid);
+        return geopm::MSRIOGroup::msr_whitelist(cpuid);
+    }
+
+    std::vector<std::string> Admin::dup_keys(const std::map<std::string, std::string> &map_a,
+                                             const std::map<std::string, std::string> &map_b)
+    {
+        std::vector<std::string> result;
+        auto a_it = map_a.begin();
+        auto b_it = map_b.begin();
+        while (a_it != map_a.end() && b_it != map_b.end()) {
+            if (a_it->first == b_it->first) {
+                result.push_back(a_it->first);
+                ++a_it;
+                ++b_it;
+            }
+            else if (a_it->first < b_it->first) {
+                ++a_it;
+            }
+            else {
+                ++b_it;
+            }
         }
         return result;
     }
 
     std::string Admin::check_node(void)
     {
-        throw geopm::Exception("Check has not been implemented yet",
-                               GEOPM_ERROR_NOT_IMPLEMENTED, __FILE__, __LINE__);
-        return "";
+        std::map<std::string, std::string> default_map =
+            Environment::parse_environment_file(m_default_config_path);
+        std::map<std::string, std::string> override_map =
+            Environment::parse_environment_file(m_override_config_path);
+        // Check for parameters that are defined in both files
+        std::vector<std::string> overlap = dup_keys(default_map, override_map);
+        if (overlap.size()) {
```

Comment:
`overlap.size() > 0`

Reply:
overlap.size() > 0U?

tomato tomato.  I can change, but I don't like it.